### DISCUSSION TITLE
HMIP HAP - Home Assistant add on

### DISCUSSION THREAD

**Discussion by ronaldt80** [OpenCCU/OpenCCU#1301] [🇺🇸 Q&A]
Hi @jens-maus ,

first of all congratulations on the great tool you built; I just migrated from debmatic on a pi4 to raspberrymatic on HaOS Odroid N2+; migration was a breeze. I did however run into an issue with one of my Hmip HAP lan gateways. I used to have two operational on Debmatic, after migration one refused to connect - the other one functions perfectly. I reset the HAP, however so far have been unable to teach it in (consistently get the dark blue led..). Also with the spare one I have laying around I did not get far.

Do I need to open any ports on the add on to the host? At present I have 2001, 2010, 43439, 9292 and 9293 open. Any other tips and tricks?

cheers,Ronald

**Comment by jens-maus** [comment]
No sorry, AFAIK no other dependencies or port exists that you need to open up. At least I dunno any other port that might have to be opened up in case you are using more than one HmIP-HAP as a LAN gateway.

**Comment by schwarzbach** [comment]
Hi,

i got the same behaviour. Migrated from RaspberryMatic directly on Charly to RaspberryMatic addon on Hassos 6.0 on the same Charly.
Restored the backup from the previous setup and all devices worked fine. Including the HMIP-HAP, which showed up on the homepage and showed values for DC und CS. 
After a while i realised, that the HMIP-HAP was not connected. After powercycling the HAP and no changes I hard removed the HAP from the CCU setup, reset it, and tried to learn it again in RaspberryMatic. But no luck so far. Tried exposing all ports in the addon, but no change. Can't find the HAP in Raspberry-Matic.

@jens-maus: Any ideas on how to debug this further?

regards,
Michael

**Comment by jens-maus** [comment]
Perhaps someone with a working HmIP-HAP connection can debug with `wireshark` & co which ports are actually used while the HAP is connected. My best guess is, that perhaps there is a port missing and which needs to be exposed to let the HAP work correctly in the HA add-on situation.

**Reply by schwarzbach** [reply]
ich werde in den kommenden Tagen mal versuchen, RaspberryMatic wieder direkt auf dem Charly zu starten und den HAP wieder einzubinden. Und dabei mal den Traffic mitzuschneiden. Mal sehen, ob hierbei dann noch andere Ports eine Rolle spielen.

**Reply by schwarzbach** [reply]
so, booted RaspberryMatic directly (old SD-card image, where the HAP was still configured as a LAN gateway) and the HAP immediately came up as a LAN gateway. Sniffing the traffic while RaspberryMatic booted was not successful yet, but i had a look at RaspberryMatic via SSH. All open ports are already configured in the add-on.

**Reply by ronaldt80** [reply]
Could it be igmp snooping related? I disabled this on my switches (as it is known to cause issues with hmip hap) but seeing this is a container, could it be that the docker network is causing issues?

**Comment by oekinger** [comment]
Hi,

same story as I switched from Tinker Board to Home Assistant. One of my HMIP-HAP is connected and seems to work, the other one can't be connected any more. It seems, that only one HMIP-HAP can work at the same time, so it could be a port-related problem (?).

Sorry if I with the language: Komisch ist auch, dass der funktionierende HMIP-HAP türkis blinkt, nicht dauerhaft leuchtet. Auch ich habe IGMP-Snooping deaktiviert, ebenso drop von unbekannten Multicast-Paketen. Allerdings hab ich die Switch auch mit IGMP Shopping getestet: es macht keinen Unterschied.

If I could assist in any way: please advise what to do.

Greetings,
Marko
